implementation/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/TypeResolver.java

```diff
@@ -76,37 +203,201 @@ public Type getResolvedType(Type fieldType) {
-                fields.put(field, resolver);
-            }
+            // Store all field properties
+            currentClass.fields()
+                    .stream()
+                    .filter(field -> !Modifier.isStatic(field.flags()))
+                    .forEach(field -> scanField(properties, field, stack));
+
+            currentClass.methods()
+                    .stream()
+                    .filter(method -> !Modifier.isStatic(method.flags()))
+                    .forEach(method -> scanMethod(properties, method, stack));
+
+            currentClass.interfaceTypes()
+                    .stream()
+                    .map(index::getClass)
+                    .filter(Objects::nonNull)
+                    .flatMap(clazz -> clazz.methods().stream())
+                    .forEach(method -> scanMethod(properties, method, stack));
+        }
+
+        return sorted(properties, chain.keySet());
+    }
+
+    private static Map<ClassInfo, Type> inheritanceChain(AugmentedIndexView index,
+            ClassInfo klazz,
+            Type type) {
+
+        Map<ClassInfo, Type> chain = new LinkedHashMap<>();
+
+        do {
+            chain.put(klazz, type);
+        } while ((type = klazz.superClassType()) != null && (klazz = index.getClass(type)) != null);
+
+        return chain;
+    }
 
-            currentType = currentClass.superClassType();
+    private static void scanField(Map<String, TypeResolver> properties, FieldInfo field, Deque<Map<String, Type>> stack) {
+        String propertyName = field.name();
+        String methodSuffix = Character.toUpperCase(propertyName.charAt(0)) + propertyName.substring(1);
 
-            if (currentType == null) {
-                break;
+        // Consider only using fields that are public?
+        if (properties.containsKey(propertyName)) {
+            TypeResolver resolver = properties.get(propertyName);
+
+            if (resolver.getField() == null && (Modifier.isPublic(field.flags()) || Modifier.isProtected(field.flags()))) {
+                /*
+                 * Field is declared in parent class and the getter/setter was
+                 * declared/overridden in child class
+                 */
+                resolver.setField(field);
             }
+        } else {
+            TypeResolver resolver = new TypeResolver(field, new ArrayDeque<>(stack));
+            ClassInfo declaringClass = field.declaringClass();
+            resolver.setReadMethod(declaringClass.method("get" + methodSuffix));
+            resolver.setWriteMethod(declaringClass.method("set" + methodSuffix, field.type()));
+            properties.put(propertyName, resolver);
+        }
+    }
+
+    private static void scanMethod(Map<String, TypeResolver> properties, MethodInfo method, Deque<Map<String, Type>> stack) {
+        String methodName = method.name();
+        Type returnType = method.returnType();
+        Type propertyType;
+
+        if (methodName.startsWith("get") &&
```

Comment:
The javabeans convention for booleans is different so better include this as well, e.g.
`if (methodName.startsWith("get") || methodName.startsWith("is"))`
The substring(4) trick will no longer work as well :)

Reply:
Good catch, I overlooked that.

Reply:
Added support for boolean methods name `isXXX`.